sources/Inputs/src/AutomatonConfigurationManager.cpp

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  AutomatonConfigurationManager.cpp
+ *
+ * @brief Automaton manager implementation file
+ *
+ */
+
+#include "AutomatonConfigurationManager.h"
+
+#include "Log.h"
+#include "Message.hpp"
+
+#include <boost/filesystem.hpp>
+#include <fstream>
+#include <iostream>
+#include <xml/sax/parser/ComposableDocumentHandler.h>
+#include <xml/sax/parser/Parser.h>
+#include <xml/sax/parser/ParserException.h>
+#include <xml/sax/parser/ParserFactory.h>
+
+namespace parser = xml::sax::parser;
+namespace file = boost::filesystem;
+
+namespace dfl {
+namespace inputs {
+
+namespace helper {
+
+template<class T>
+/**
+ * @brief Trait class for xsd filename by type
+ */
+struct XsdTrait {
+  static const std::string filename;  ///< filename trait definition, to specialize
+};
+
+template<>
+const std::string XsdTrait<AssemblyXmlDocument>::filename("assembling.xsd");  ///< XSD filename for assembly xml document
+template<>
+const std::string XsdTrait<SettingsXmlDocument>::filename("setting.xsd");  ///< XSD filename for setting xml document
+
+template<class T>
+static file::path
+computeXsdPath(const file::path& filepath) {
+  auto basename = filepath.filename().replace_extension().generic_string();
+
+  auto var = getenv("DYNAFLOW_LAUNCHER_XSD");
+  file::path xsdFile;
```

Comment:
I would name this one xsdFolder

Reply:
In fine, xsdFile represents the xsd filepath, not only the folder